csharp/src/Apache.Arrow.Adbc/Interop/NativePointer.cs

```diff
@@ -0,0 +1,39 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Apache.Arrow.Adbc.Interop
+{
+    struct NativePointer<T>
+    {
+        readonly T managedDelegate; // For lifetime management
+        readonly IntPtr nativePointer;
+
+        public NativePointer(T managedDelegate)
+        {
+            this.managedDelegate = managedDelegate;
```

Comment:
Can this code use the same style as apache/arrow/csharp uses? Basically this:

https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md